File: MobiTaskClient/app/src/main/java/com/task/mobi/data/RouteData.java
package com.task.mobi.data;

/**
 * Created by neeraj on 3/10/17.
 */


import android.os.Parcel;
import android.os.Parcelable;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import java.io.Serializable;

public class RouteData implements Serializable, Parcelable {

    public final static Creator<RouteData> CREATOR = new Creator<RouteData>() {


        @SuppressWarnings({
                "unchecked"
        })
        public RouteData createFromParcel(Parcel in) {
            return new RouteData(in);
        }

        public RouteData[] newArray(int size) {
            return (new RouteData[size]);
        }

    };
    @SerializedName("latitude")
    @Expose
    public Double lat;
    @SerializedName("longitude")
    @Expose
    public Double lng;

    @SerializedName("address")
    @Expose
    public String address;

    @SerializedName("date")
    @Expose
    public String date;

    public String distance;

    protected RouteData(Parcel in) {
        this.lat = ((Double) in.readValue((Double.class.getClassLoader())));
        this.lng = ((Double) in.readValue((Double.class.getClassLoader())));
        this.distance = ((String) in.readValue((String.class.getClassLoader())));
        this.address = ((String) in.readValue((String.class.getClassLoader())));
        this.date = ((String) in.readValue((String.class.getClassLoader())));
    }

    public void writeToParcel(Parcel dest, int flags) {
        dest.writeValue(lat);
        dest.writeValue(lng);
        dest.writeValue(distance);
        dest.writeValue(address);
        dest.writeValue(date);
    }

    public int describeContents() {
        return 0;
    }

}

File: MobiTaskAdmin/app/src/main/java/com/task/mobiadmin/data/RouteData.kt
package com.task.mobiadmin.data

/**
 * Created by neeraj on 3/10/17.
 */


import android.os.Parcel
import android.os.Parcelable
import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

data class RouteData(
        @SerializedName("latitude") @Expose var lat: Double = 0.0,
        @SerializedName("longitude") @Expose var lng: Double = 0.0,
        @SerializedName("address") @Expose var address: String,
        @SerializedName("date") @Expose var date: String,
        var distance: String) : Parcelable {
    constructor(parcel: Parcel) : this(
            parcel.readDouble(),
            parcel.readDouble(),
            parcel.readString(),
            parcel.readString(),
            parcel.readString()) {
    }

    override fun writeToParcel(parcel: Parcel, flags: Int) {
        parcel.writeDouble(lat)
        parcel.writeDouble(lng)
        parcel.writeString(address)
        parcel.writeString(date)
        parcel.writeString(distance)
    }

    override fun describeContents()= 0

    companion object CREATOR : Parcelable.Creator<RouteData> {
        override fun createFromParcel(parcel: Parcel): RouteData {
            return RouteData(parcel)
        }

        override fun newArray(size: Int): Array<RouteData?> {
            return arrayOfNulls(size)
        }
    }
}